### ISSUE TITLE
Show Device Name for Top 10 Blocked Devices (Outbound)

### ISSUE THREAD

**Issue by kramttocs** [Adamm00/IPSet_ASUS#107]
I searched but didn't get a hit for this.

In the Top 10 Blocked Devices (Outbound) table I see the IP and then "(Unknown)"

In the Client List I have this device named similar to "Wife iPhone" but I do not have this set to a static IP so it's not in the DHCP list.
Is there a way to replace "(Unknown)" with "(Wife iPhone)"?

Thank you

**Comment by kramttocs** [comment]
I meant to add that I did find this: https://www.snbforums.com/threads/skynet-router-firewall-security-enhancements.16798/post-547268

Not sure what to do with that though.

**Comment by Adamm00** [comment]
The webui gets device names in a way that isn’t possible for us. Unless someone has a creative solution this is as good as it gets.

**Comment by jumpsmm7** [comment]
> The webui gets device names in a way that isn’t possible for us. Unless someone has a creative solution this is as good as it gets.

When you say it gets the device name? How do you suspect it is doing this - arp requests, arp cache, or nslookup?

**Comment by Adamm00** [comment]
So I had one of those "how did I never consider this" moments recently when being frustrated by lack of population on the client list. Give https://github.com/Adamm00/IPSet_ASUS/commit/c9ad3e8c2bd6004250e3d94748ddde7f7eee0d76 and see how things improve.

**Comment by kramttocs** [comment]
That did make all of them 'fill in'. Thanks!
I am not sure it's quite perfect yet :) 
For some devices, it's not showing the name that shows up in the client list but rather a manufacturers name (like AsusTek Computer Inc and Plugable [this is for my laptop dock]). Both of these do have more 'human' names in the client list that I've assigned them. Before this change, I am sure the Asus one showed correctly though I can't recall about my laptop.
Yet other devices like my wife's phone now is showing up with the name I gave it so that's great. Odd
